python/ctsm/toolchain/gen_mksurfdata_namelist.py

```diff
@@ -491,271 +882,45 @@ def main():
-
-    # ----------------------------------------
-    # Write output namelist file
-    # ----------------------------------------
-
-    with open(nlfname, "w", encoding="utf-8") as nlfile:
-        nlfile.write("&mksurfdata_input \n")
-
-        # -------------------
-        # raw input data
-        # -------------------
-        if force_model_mesh_file is None:
-            mksrf_fgrid_mesh_nx = rawdata_files["mksrf_fgrid_mesh_nx"]
-            mksrf_fgrid_mesh_ny = rawdata_files["mksrf_fgrid_mesh_ny"]
-            mksrf_fgrid_mesh = rawdata_files["mksrf_fgrid_mesh"]
-        else:
-            mksrf_fgrid_mesh_nx = force_model_mesh_nx
-            mksrf_fgrid_mesh_ny = force_model_mesh_ny
-            mksrf_fgrid_mesh = force_model_mesh_file
-        nlfile.write(f"  mksrf_fgrid_mesh = '{mksrf_fgrid_mesh}' \n")
-        nlfile.write(f"  mksrf_fgrid_mesh_nx = {mksrf_fgrid_mesh_nx} \n")
-        nlfile.write(f"  mksrf_fgrid_mesh_ny = {mksrf_fgrid_mesh_ny} \n")
-
-        for key, value in rawdata_files.items():
-            if key == "mksrf_ftopostats" and mksrf_ftopostats_override != "":
-                nlfile.write(f"  mksrf_ftopostats_override = '{mksrf_ftopostats_override}' \n")
-            elif "_fvic" not in key and "mksrf_fvegtyp" not in key and "mksrf_fgrid" not in key:
-                # write everything else
-                nlfile.write(f"  {key} = '{value}' \n")
-
-        if start_year <= 2015:
-            mksrf_fvegtyp = rawdata_files["mksrf_fvegtyp"]
-            mksrf_fvegtyp_mesh = rawdata_files["mksrf_fvegtyp_mesh"]
-            mksrf_fhrvtyp = rawdata_files["mksrf_fvegtyp"]
-            mksrf_fhrvtyp_mesh = rawdata_files["mksrf_fvegtyp_mesh"]
-            mksrf_fpctlak = rawdata_files["mksrf_fvegtyp_lake"]
-            mksrf_furban = rawdata_files["mksrf_fvegtyp_urban"]
-        else:
-            mksrf_fvegtyp = rawdata_files["mksrf_fvegtyp_ssp"]
-            mksrf_fvegtyp_mesh = rawdata_files["mksrf_fvegtyp_ssp_mesh"]
-            mksrf_fhrvtyp = rawdata_files["mksrf_fvegtyp_ssp"]
-            mksrf_fhrvtyp_mesh = rawdata_files["mksrf_fvegtyp_ssp_mesh"]
-            mksrf_fpctlak = rawdata_files["mksrf_fvegtyp_ssp_lake"]
-            mksrf_furban = rawdata_files["mksrf_fvegtyp_ssp_urban"]
-        if "%y" in mksrf_fvegtyp:
-            mksrf_fvegtyp = mksrf_fvegtyp.replace("%y", str(start_year))
-        if "%y" in mksrf_fhrvtyp:
-            mksrf_fhrvtyp = mksrf_fhrvtyp.replace("%y", str(start_year))
-        if "%y" in mksrf_fpctlak:
-            mksrf_fpctlak = mksrf_fpctlak.replace("%y", str(start_year))
-        if "%y" in mksrf_furban:
-            mksrf_furban = mksrf_furban.replace("%y", str(start_year))
-        if not os.path.isfile(mksrf_fvegtyp):
-            print("WARNING: input mksrf_fvegtyp file " f"{mksrf_fvegtyp} does not exist")
-            print("WARNING: run ./download_input_data to try TO " "OBTAIN MISSING FILES")
-            _must_run_download_input_data = True
```

Comment:
You got rid of the leading underscore here. And I think that makes sense here. That's only when it's meant to be private and since this is now what's returned for the new function -- this makes sense.